dot/network/state.go

```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/ChainSafe/gossamer/lib/common"
 )
 
+//go:generate mockgen -destination=mock_block_state_interface.go -package $GOPACKAGE . BlockState
```

Comment:
Updated.

Reply:
For some reason when I change this, I can no longer get reference to `network.NewMockBlockState(ctrl)` in node_test.go

Reply:
This is because test.go files are not available to other packages. Instead let's have some mock duplication to reduce package coupling due to testing.

So in whichever package you need `network.NewMockBlockState(ctrl)`, add a

```go
//go:generate mockgen -destination=mock_block_state_test.go -package $GOPACKAGE github.com/ChainSafe/gossamer/dot/network BlockState
```

Or perhaps define a `BlockState` interface will less methods than the one in `dot/network`, and generate a mock from that one if you can.